src/android/UploadTask.java

```diff
@@ -457,4 +515,15 @@ public void writeTo(@NonNull BufferedSink bufferedSink) throws IOException {
             }
         }
     }
+
+    private class NetworkReceiver extends BroadcastReceiver {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+            if((connectivityManager == null) || (connectivityManager.getActiveNetworkInfo() == null) || (connectivityManager.getActiveNetworkInfo().isConnected() == false)) {
+                Log.d(TAG, "WIFI is not connected");
```

Comment:
does this network check only cater for WIFI connection ? does it take into account mobile data ?

Reply:
Yes it check for any network connection

Reply:
so a better log message could be  "No internet connection" ? @zafirskthelifehacker

Reply:
@dinitri-ragoo yeps